hail/src/main/scala/is/hail/compatibility/LegacyRVDSpecs.scala

```diff
@@ -32,12 +33,12 @@ case class IndexSpec private(
       EField("annotation", annotationEType, 2)
     ), required = true), required = true), 1)
   ))
-  val leafVType = TStruct(FastIndexedSeq(
-    Field("first_idx", TInt64Required, 0),
-    Field("keys", TArray(TStruct(FastIndexedSeq(
-      Field("key", keyVType, 0),
-      Field("offset", TInt64Required, 1),
-      Field("annotation", annotationVType, 2)
+  val leafPType = PStruct(FastIndexedSeq(
```

Comment:
Necessary changes to RVD, BgenRDD, TypedCodecSpec, AbstractCodecSpec to thread PTypes through, since we should not carry requiredeness on virtual types. This also allows ReadPartition to get the correct physical type.